src/Engine/IEngine.cs

```diff
@@ -2,7 +2,7 @@
 
 namespace CleanLiving.Engine
 {
-    public interface IEngine<TTime>
+    public interface IEngine<TTime> : IDisposable
```

Comment:
I went with this due to the reason in the issue #23, assuming all implementations will in some way hold onto disposables. I guess they might not though as they may get passed off into another dependency or something in a future implementation/this implementation gets changed?